documentation/puppetserver_vs_passenger.markdown

```diff
@@ -70,3 +70,22 @@ logged using `Puppet.info` will be logged in Puppet Server and would not have
 been under Passenger. See
 [Puppet Server Configuration Files: logback.xml](./config_file_logbackxml.md)
 for more information, as well as instructions on how to set log levels in Puppet Server.
+
+
+## External Command Execution
+
+When writing extensions that target the server, extension authors should be
+careful to use Puppet's built in execution helper: `Puppet::Util::Execution.execute`
+rather than native Ruby methods (eg "backticks", `system`, or `spawn`).
+
+Ruby's native methods for spawning processes will cause a fork of the JVM on
+most Linux servers, which in a large production environment can easily cause
+Out of Memory errors at the OS level. Puppet Server provides a lighter weight
+way of creating sub-processes which it exposes through
+`Puppet::Util::Execution.execute`.
+
+You should use this during development of all extensions unless there are
+better methods available. For example you should use
+`Puppet::Util::Execution.execute` when writing Ruby based functions, custom
+report processors, Hiera backends, and faces. When writing custom providers
+the `command` helper should be used if possible to help determine suitability.
```

Comment:
This sentence is hard for me to understand. 1) Is there a link we can provide explaining what the `command` helper is? 2) What does "to help determine suitability" mean? I'm probably just missing some knowledge here, but other people might be too.

Reply:
> 1) Is there a link we can provide explaining what the command helper is?

I'm not sure what the `command` helper refers to here, but if it's `Puppet::Provider::Command` or something in the `Puppet::Util::Execution` YARD docs, then it'd have to be to rubydoc.info. http://www.rubydoc.info/gems/puppet/Puppet/Provider/Command

> 2) What does "to help determine suitability" mean? I'm probably just missing some knowledge here, but other people might be too.

+1